### PR TITLE
Added TypeScript definitions for css prop.

### PR DESCRIPTION
**Pull request by cameron-martin**
**What**:
Added TypeScript definitions for the `css` prop which is compiled to `className` by the babel plugin.

**Why**:
Because otherwise TypeScript users will get type errors when using this feature.

**How**:
Used module augmentation/declaration merging to add the `css` property to the `HTMLAttributes` inferface.

**Checklist**:

- [ ] Documentation
- [x] Tests
- [x] Code complete

### CODE REVIEW

**Comment by cameron-martin** [comment]
On seconds thoughts, maybe this should go into the dts files in the `emotion` package instead, since the css prop can be used without using the `react-emotion` package. Or even in the types for the `babel-plugin-emotion` package - although I have no idea if this would work.

What do you think?

**Comment by tkh44** [comment]
I should go in the react-emotion package I think.

File: packages/react-emotion/typings/react-emotion.d.ts [hunk 1]
```diff
@@ -3,6 +3,12 @@ import { Interpolation as EmotionInterpolation } from 'emotion'
 
 export * from 'emotion'
 
+declare module 'react' {
+  interface HTMLAttributes<T> {
+    css?: string;
```
Comment (tkh44) [inline packages/react-emotion/typings/react-emotion.d.ts]: This can be any type of value.

Comment (cameron-martin) [reply packages/react-emotion/typings/react-emotion.d.ts]: Anything of type `Interpolation`?

**Comment by cameron-martin** [comment]
I'm using react + emotion without react-emotion so it would be useful to be in the emotion package.

File: packages/react-emotion/typings/react-emotion.d.ts (hunk 1 above)
Comment (tkh44) [reply packages/react-emotion/typings/react-emotion.d.ts]: Correct

**Comment by tkh44** [comment]
I agree. @cameron-martin. I misread your original comment. Sorry about that.

**Comment by cameron-martin** [comment]
I've made the css prop take anything of type `Interpolation` and moved it to the emotion package.

**Pull request merged**